File: java/src/main/java/io/fivexer/sdk/model/PatchSkill.java
package io.fivexer.sdk.model;

import io.fivexer.sdk.internal.Json;

/**
 * Partial skill update — the {@code key} is immutable.
 *
 * <p>Fluent builder: set what you need, then hand it to the client. {@link #toJson()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
public class PatchSkill {
    private String name;
    private String description;

    public PatchSkill() {}

    public String getName() { return name; }
    public String getDescription() { return description; }

    public PatchSkill name(String name) { this.name = name; return this; }
    public PatchSkill description(String description) { this.description = description; return this; }

    /** Serialise for the wire, omitting unset fields. */
    public String toJson() {
        return Json.write(this);
    }
}

File: php/src/Model/PatchSkill.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Partial skill update - the `key` is immutable.
 *
 * Fluent builder: set what you need, then hand it to the client. {@see self::toArray()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
final class PatchSkill
{
    public function __construct(
        private ?string $name = null,
        private ?string $description = null,
    ) {
    }

    public function name(string $name): self
    {
        $this->name = $name;
        return $this;
    }

    public function description(string $description): self
    {
        $this->description = $description;
        return $this;
    }

    /**
     * Serialise for the wire, omitting unset fields.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        $payload = Json::compact([
            'name' => $this->name,
            'description' => $this->description,
        ]);
        return $payload;
    }
}
